### PR TITLE
added functionality to add club members and display info

### PR DESCRIPTION
**Pull request by Gauravtb2253**
Hello
I have added the required functionality
Pls check

### CODE REVIEW

**Comment by Yash-Ainapure** [comment]
@Gauravtb2253 ,the build is failing,
before pushing do "npm run build" for both frontend & backend folder to check for any typescript errors and remove them

File: frontend/src/appComponents/axiosInstance.tsx
Comment (Yash-Ainapure) [inline frontend/src/appComponents/axiosInstance.tsx:1]: change this back to backend url

**Comment by Gauravtb2253** [comment]
so I thought in hiring we have to add member

**Comment by Gauravtb2253** [comment]
Like can you check the current implementation and tell what to chaneg

File: frontend/src/appComponents/HiringPage.tsx
[the remark this answers is not in the document]
Comment (Yash-Ainapure) [reply frontend/src/appComponents/HiringPage.tsx:1]: Create a new page for adding club member and add a button on admin panel's dashboard to add a new member which will redirect to that page which you will create to add members.

and also put the old code of hiringPage.tsx as same from:
https://github.com/Yash-Ainapure/dypcetclubs.live/blob/main/frontend/src/appComponents/HiringPage.tsx

File: frontend/src/App.tsx
Comment (Yash-Ainapure) [inline frontend/src/App.tsx:1]: Also please verify this file,as some errors are present here

**Comment by Gauravtb2253** [comment]
what is the exact flow for adding members you need can you tell

**Comment by Yash-Ainapure** [comment]
> what is the exact flow for adding members you need can you tell

The club admin login's himself and gets redirected to the admin dashboard,where he sees the dashboard page a button to add new club members.
on clicking on add new member , he would get redirected to a page to get input for new member info and a add button to add new member.


In further issues, we can work on other dashboard things and how to update members,etc
Be free to ask any questions if required

**Comment by Gauravtb2253** [comment]
So currently I added the process in the hiring page
So I have to shift it to a new page for adding members
Right?

**Comment by Yash-Ainapure** [comment]
> So currently I added the process in the hiring page So I have to shift it to a new page for adding members Right?

Yes

**Comment by Gauravtb2253** [comment]
image

should I add the button here

**Comment by Yash-Ainapure** [comment]
> image
> 
> should I add the button here

No,
when we click on dashboard, there is a demo page, add the button there.

**Comment by Gauravtb2253** [comment]
Okk

**Comment by Yash-Ainapure** [comment]
> Okk

The file of demo dashboard is present in
ClubAdmin.tsx Dashboard component

**Comment by Gauravtb2253** [comment]
Yes got it

**Comment by Gauravtb2253** [comment]
pls check and let me know if changes

**Comment by Gauravtb2253** [comment]
Pls let me know if you have more issues

**Comment by Yash-Ainapure** [comment]
Thanks for your contributions.

**Comment by Gauravtb2253** [comment]
Hello any updates for more issues

**Comment by Yash-Ainapure** [comment]
> Hello any updates for more issues

Will tell you till tomorrow

**Comment by Gauravtb2253** [comment]
hello any updates

**Comment by Gauravtb2253** [comment]
hello

**Comment by Yash-Ainapure** [comment]
> hello

Heyy,Sorry for delay.
within few mins, i will push backend logic for hiring things. the oyu could do the whole frontend part of that.
are you ok with that?

**Comment by Gauravtb2253** [comment]
yes fine

**Comment by Gauravtb2253** [comment]
just tell me how should be the flow of UI

**Comment by Yash-Ainapure** [comment]
> just tell me how should be the flow of UI

(create a new issue for this)
First we will create admin side->
on hiring tab of admin dashboard create a button to create a new hiring session(routes are added at backend), which will open a square modal on same page to get the required inputs and creates a new session.
also fetch already created sessions by that club only to display.(add a delete button to delete that,if possible update button also.

when we create a session it gets listed there and by clicking on that session will open a new page on dashboard(like view event) where we can add/update/delete hiring positions.

also here we have to add  view applications thing where we can view all applications done by the students.



user side->
and then on the the user side on hiring tab we will list all the hiring sessions by clubs where user choses a session and choose a position,fill required details and submit the application.


this whole things could take multiple issues as a separate just make them as you wish (it will be better to give me points) like that.

**Comment by Yash-Ainapure** [comment]
Create new issue for this and start working

**Comment by Gauravtb2253** [comment]
First we will create admin side->
on hiring tab of admin dashboard create a button to create a new hiring session(routes are added at backend), which will open a square modal on same page to get the required inputs and creates a new session.
also fetch already created sessions by that club only to display.(add a delete button to delete that,if possible update button also.


should I first create a issue for above

**Comment by Yash-Ainapure** [comment]
> First we will create admin side-> on hiring tab of admin dashboard create a button to create a new hiring session(routes are added at backend), which will open a square modal on same page to get the required inputs and creates a new session. also fetch already created sessions by that club only to display.(add a delete button to delete that,if possible update button also.
> 
> should I first create a issue for above

Sure

**Pull request merged**